pyuvm/s25_pyuvm_adapter.py

```diff
@@ -0,0 +1,92 @@
+## Import Main Package
+from pyuvm import uvm_object
+from pyuvm import uvm_sequence_item
+from pyuvm import uvm_sequence
+from pyuvm.s24_pyuvm_reg_includes import *
+from pyuvm.s23_pyuvm_reg_item import *
+
+## Main Class
+class pyuvm_reg_adapter(uvm_object):
+    ## Constructor
+    def __init__(self, name="pyuvm_reg_adapter"):
+        super().__init__(name)
+        #    Set this bit in extensions of this class if the bus protocol supports
+        #    byte enables.
+        self.byte_enable    = True
+        self.parent_sequence= uvm_sequence  
+        self.reg_item       = uvm_sequence_item 
+
+    #    Function -- reg2bus
+    #    Extensions of this class ~must~ implement this method to convert the specified
+    #    <uvm_reg_bus_op> to a corresponding <uvm_sequence_item> subtype that defines the bus
+    #    transaction.
+    #    The method must allocate a new bus-specific <uvm_sequence_item>,
+    #    assign its members from
+    #    the corresponding members from the given generic ~rw~ bus operation, then
+    #    return it.
+    def reg2bus(rw: uvm_reg_bus_op):
+        pass
```

Comment:
Shouldn't this be `raise <exception>` instead of `pass` ? User must implement `reg_adapter`

Reply:
> Shouldn't this be `raise <exception>` instead of `pass` ? User must implement `reg_adapter`

Yes.  It should raise an exception unless`pass` is a reasonable behavior.  There is also `error_classes.UVMNotImplemented()` if you want a more descriptive exception.

Reply:
Correct should raise an exception is done at my end I just have to push it thanks for highlighting it